src/Symfony/Component/Mailer/Transport/AbstractTransport.php

```diff
@@ -38,7 +38,7 @@ abstract class AbstractTransport implements TransportInterface
     public function __construct(EventDispatcherInterface $dispatcher = null, LoggerInterface $logger = null)
     {
         $this->dispatcher = $dispatcher;
-        $this->logger = $logger ?? new NullLogger();
+        $this->setLogger($logger ?? new NullLogger());
```

Comment:
no need to use the setter here

Reply:
I just added it, psalm throws error without that, and it seems this PR is going to be closed

Reply:
what do you mean by "it throws error" ? The existing code `$this->logger = $logger ?? new NullLogger();` should be totally fine.

Reply:
Yes, it is fine, error is "`setLogger` method is never used"

Reply:
Yes, it is fine, error is "setLogger method is never used"
Would it be merge if I returned it to how it was initially?

Reply:
Don't care about this one. We should change our psalm config to disable those dead code detection rules.

The method being added is expected to be part of the public API, so we don't care that it is never used in Symfony itself.

Reply:
OK, thanks for reviewing